Mathlib/LinearAlgebra/Finsupp/Sym2.lean

```diff
@@ -0,0 +1,52 @@
+/-
+Copyright (c) 2025 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+
+import Mathlib.Data.Sym.Sym2
+import Mathlib.Algebra.Ring.Defs
+import Mathlib.Data.Finset.Sym
+import Mathlib.Data.Finsupp.Defs
+
+/-!
+# Finitely Supported Commutative multiplication
+
+-/
+
+section
+
+variable {α R}
+
+/-- A commutative magma with multiplicative zero. -/
+class CommMagmaMulZeroClass (R : Type*) extends CommMagma R, MulZeroClass R
```

Comment:
Do you need this new class? Do you care about an example which is not a `CommMonoidWithZero`?

Reply:
No, thanks, `CommMonoidWithZero` is perfect. (I had `CommSemiring` before which felt too specific.)